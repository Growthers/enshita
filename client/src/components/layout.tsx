import React from "react";
import Header from "~/components/header"
import Footer from "~/components/footer"
import style from "~/styles/layout.module.scss"

type Props = {
  children: React.ReactNode
}

const Layout: React.FC<Props> = ({children}) => (
  <div>
    <Header/>
    <div className={style["enshita-layout-body"]}>
      {children}
    </div>
    <Footer/>
  </div>
)

export default Layout
