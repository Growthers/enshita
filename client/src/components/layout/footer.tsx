import React from "react";
import styles from "~/styles/components/layout/footer.module.scss";

const Footer: React.FC = () => (
  <footer>
    <div className={styles["enshita-footer"]}>
      <div className={styles["enshita-footer-content"]}>
        <div
          className={`${styles["enshita-footer-logo"]} ${styles["enshita-footer-hidden"]}`}
        >
          <svg
            width="140"
            height="50"
            xmlns="http://www.w3.org/2000/svg"
            viewBox="0 0 140 50"
          >
            <text
              x="0"
              y="40"
              fontSize={40}
              className={styles["enshita-footer-svg"]}
            >
              enshita
            </text>
          </svg>
        </div>
        <div className={styles["enshita-footer-menu"]}>
          <p className={styles["enshita-footer-about"]}>About Growthers</p>
          <ul className={styles["enshita-footer-list"]} role="navigation">
            <li className={styles["enshita-footer-homepage"]}>
              <a
                href="~/components/layout/footer"
                className={`${styles["enshita-footer-link"]}`}
                target="_blank"
                rel="noopener noreferrer"
              >
                Home Page
              </a>
            </li>
            <li className={styles["enshita-footer-twitter"]}>
              <a
                href="~/components/layout/footer"
                className={`${styles["enshita-footer-link"]}`}
                target="_blank"
                rel="noopener noreferrer"
              >
                Twitter
              </a>
            </li>
            <li className={styles["enshita-footer-github"]}>
              <a
                href="~/components/layout/footer"
                className={`${styles["enshita-footer-link"]}`}
                target="_blank"
                rel="noopener noreferrer"
              >
                GitHub
              </a>
            </li>
          </ul>
        </div>
      </div>
      <p className={styles["enshita-footer-copyright"]}>
        <small>Copyright &copy; 2022 Growthers</small>
      </p>
    </div>
  </footer>
);

export default Footer;
