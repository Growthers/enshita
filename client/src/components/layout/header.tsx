import type { FC } from "react";
import styles from "~/styles/components/layout/header.module.scss";
import { Link } from "wouter";
import AnchorButton from "../button/anchor-button";
import type { HeaderProperties } from "./type/model";

const Header: FC<HeaderProperties> = ({ disableStyle }) => (
  <header className={`${styles["enshita-header"]} ${disableStyle}`}>
    <div>
      <Link to="/">
        <img
          src="/statics/lyrics-dark.svg"
          alt="enshita logo"
          className={styles["enshita-header-logo-img"]}
        />
      </Link>
    </div>
    <div className={styles["enshita-header-hidden"]}>
      <AnchorButton
        variant="normal"
        href="./login"
        boxStyles={styles["enshita-header-button"]}
      >
        login
      </AnchorButton>
    </div>
  </header>
);

export default Header;
