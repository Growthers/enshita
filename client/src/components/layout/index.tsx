import type { FC } from "react";
import Header from "~/components/layout/header";
import Footer from "~/components/layout/footer";
import style from "~/styles/components/layout/layout.module.scss";
import { Fragment } from "react";
import { useRoute } from "wouter";
import { LayoutProperties } from "./type/model";

const Layout: FC<LayoutProperties> = ({ children }) => {
  const [match] = useRoute("/stream/:anything*");
  return (
    <Fragment key="layout">
      <Header disableStyle={match ? style["enshita-layout-disable"] : ""} />
      <main className={match ? "" : style["enshita-layout-body"]}>
        {children}
      </main>
      <Footer disableStyle={match ? style["enshita-layout-disable"] : ""} />
    </Fragment>
  );
};

export default Layout;
