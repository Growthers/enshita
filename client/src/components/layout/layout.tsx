import type { FC } from "react";
import Header from "~/components/layout/header";
import Footer from "~/components/layout/footer";
import style from "~/styles/components/layout/layout.module.scss";
import {Fragment} from "react";
import {LayoutProperties} from "./type/model";


const Layout: FC<LayoutProperties> = ({ children }) => (
  <Fragment key="layout">
    <Header />
    <main className={style["enshita-layout-body"]}>
      {children}
    </main>
    <Footer />
  </Fragment>
);

export default Layout;
