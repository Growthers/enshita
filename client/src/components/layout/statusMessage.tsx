import type { FC } from "react";
import styles from "~/styles/components/layout/statusMessage.module.scss";

type StatusMessgeProps = {
  code: number;
  message: string;
};

const StatusMessge: FC<StatusMessgeProps> = ({ code, message }) => (
  <div className={styles["enshita-status-body"]}>
    <div className={styles["enshita-status-content"]}>
      <div className={styles["enshita-status-code"]}>
        <p>{code}</p>
      </div>
      <div className={styles["enshita-status-message"]}>
        <div className={styles["enshita-status-errormessage"]}>
          <p>{message}</p>
        </div>
        <div>
          <p>
            Oh... This URL is wrong.
            <br />
            please check again page URL.
          </p>
        </div>
      </div>
    </div>
  </div>
);

export default StatusMessge;
