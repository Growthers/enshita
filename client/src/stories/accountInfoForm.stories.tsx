// eslint-disable-next-line import/no-extraneous-dependencies
import { ComponentMeta, ComponentStoryObj } from "@storybook/react";
import Form from "~/components/form/accountInfoForm";

type T = typeof Form;
type Story = ComponentStoryObj<T>;

export default {
  title: "Account Information Form",
  args: {
    mail: "abc@example.com",
    userName: "hoge",
  },
  component: Form,
} as ComponentMeta<T>;

export const Default: Story = {};
