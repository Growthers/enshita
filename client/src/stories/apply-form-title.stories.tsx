import type { ComponentStoryObj, ComponentMeta } from "@storybook/react";
import { Title } from "~/components/apply-form/title";

type T = typeof Title;
type Story = ComponentStoryObj<T>;

const data = {
  title: "共同開発鯖LT #4",
};

export default { args: { title: data.title }, component: Title } as ComponentMeta<T>;

export const Default: Story = {};
