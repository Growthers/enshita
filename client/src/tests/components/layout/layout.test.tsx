import { composeStories } from "@storybook/testing-react";
import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import * as stories from "~/stories/layout/layout.stories";

const { Default } = composeStories(stories);

const options = {
  name: "",
};
describe("(components) header", () => {
  test("test Snap Shot", () => {
    const { container } = render(<Default />);
    expect(container).toMatchSnapshot();
  });
  test("to be [role=main]", () => {
    const { getByRole } = render(<Default />);
    expect(getByRole("main", options)).toBeInTheDocument();
  });
});
