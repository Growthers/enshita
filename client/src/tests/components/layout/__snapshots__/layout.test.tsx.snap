// Vitest Snapshot v1

exports[`(components) layout > take snapshot 1`] = `
<div>
  <header
    class="_enshita-header_41584d "
  >
    <div>
      <img
        alt="enshita logo"
        class="_enshita-header-logo-img_41584d"
        href="/"
        src="/statics/lyrics-dark.svg"
      />
    </div>
    <div
      class="_enshita-header-hidden_41584d"
    >
      <a
        class="_enshita-button-box_fd3da4 _enshita-header-button_41584d"
        href="./login"
        rel="noopener noreferrer"
        role="button"
        target="_blank"
      >
        <p
          class="_enshita-button-text_fd3da4 undefined"
        >
          login
        </p>
      </a>
    </div>
  </header>
  <main
    class="_enshita-layout-body_acbc67"
  >
    <div>
      hogehoge
    </div>
  </main>
  <footer
    class="_enshita-footer_1843f9 "
  >
    <div
      class="_enshita-footer-content_1843f9"
    >
      <div
        class="_enshita-footer-logo_1843f9 _enshita-footer-hidden_1843f9"
      >
        <svg
          height="50"
          viewBox="0 0 140 50"
          width="140"
          xmlns="http://www.w3.org/2000/svg"
        >
          <text
            class="_enshita-footer-svg_1843f9"
            font-size="40"
            x="0"
            y="40"
          >
            enshita
          </text>
        </svg>
      </div>
      <div
        class="_enshita-footer-menu_1843f9"
      >
        <p
          class="_enshita-footer-about_1843f9"
        >
          About Growthers
        </p>
        <ul
          class="_enshita-footer-list_1843f9"
          role="navigation"
        >
          <li
            class="_enshita-footer-homepage_1843f9"
          >
            <a
              class="_enshita-footer-link_1843f9"
              href="https://growthers.dev/"
              rel="noopener noreferrer"
              target="_blank"
            >
              Home Page
            </a>
          </li>
          <li
            class="_enshita-footer-twitter_1843f9"
          >
            <a
              class="_enshita-footer-link_1843f9"
              href="https://twitter.com/UdcGrowthers/"
              rel="noopener noreferrer"
              target="_blank"
            >
              Twitter
            </a>
          </li>
          <li
            class="_enshita-footer-github_1843f9"
          >
            <a
              class="_enshita-footer-link_1843f9"
              href="https://github.com/Growthers/"
              rel="noopener noreferrer"
              target="_blank"
            >
              GitHub
            </a>
          </li>
        </ul>
      </div>
    </div>
    <p
      class="_enshita-footer-copyright_1843f9"
    >
      <small>
        Copyright Â© 2022 Growthers
      </small>
    </p>
  </footer>
</div>
`;
